examples/nemo/scripts/acc_loops_trans.py

```diff
@@ -0,0 +1,132 @@
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Authors: S. Siso, STFC Daresbury Lab
+
+''' PSyclone transformation script showing the introduction of OpenACC loop
+directives into Nemo code. '''
+
+from utils import insert_explicit_loop_parallelism, normalise_loops, \
+    enhance_tree_information, add_profiling
+from psyclone.psyGen import TransInfo
+from psyclone.psyir.nodes import Call, Loop
+from psyclone.transformations import ACCParallelTrans, ACCLoopTrans
+from psyclone.transformations import ACCRoutineTrans
+
+PROFILING_ENABLED = True
+
+
+def trans(psy):
+    ''' Add OpenACC Parallel and Loop directives to all loops, including the
+    implicit ones, to parallelise the code and execute it in an acceleration
+    device.
+
+    :param psy: the PSy object which this script will transform.
+    :type psy: :py:class:`psyclone.psyGen.PSy`
+    :returns: the transformed PSy object.
+    :rtype: :py:class:`psyclone.psyGen.PSy`
+
+    '''
+    acc_region_trans = ACCParallelTrans(default_present=False)
+    acc_loop_trans = ACCLoopTrans()
+
+    print(f"Invokes found in {psy.name}:")
+    for invoke in psy.invokes.invoke_list:
+        print(invoke.name)
+
+        if PROFILING_ENABLED:
+            add_profiling(invoke.schedule.children)
+
+        # TODO #2317: Has structure accesses that can not be offloaded and has
+        # a problematic range to loop expansion of (1:1)
+        if psy.name.startswith("psy_obs_"):
+            print("Skipping", invoke.name)
+            continue
+
+        # TODO #1841: These files have a bug in the array-range-to-loop
+        # transformation. One leads to the following compiler error
+        # NVFORTRAN-S-0083-Vector expression used where scalar expression
+        # required, the other to an incorrect result.
+        if invoke.name in ("trc_oce_rgb", ):
+            print("Skipping", invoke.name)
+            continue
+
+        # This are functions with scalar bodies, we don't want to parallelise
+        # them, but we could:
+        # - Inine them
+        # - Annotate them with 'omp declare target' and allow to call from gpus
+
+        # TODO 2019: DDPDD in additon has a wp precision symbol that PSyclone
```

Comment:
I've had a look at #2019. Would you mind checking whether DDPDD is still a problem as it may be fixed now?

Reply:
It succeeded the integration tests when removing DDPDD from this condition.